### DISCUSSION TITLE
Is it possible to write more than 20 bytes of data?

### DISCUSSION THREAD

**Discussion by tonykwon** [innoveit/react-native-ble-manager#827] [Q&A]
Hi,

I was just wondering if it is possible to write more than 20 bytes of data to a characteristic? I am working on a `OTA Firmware` Updating feature and excluding command bytes, I can only send `9 bytes` of the firmware data at a time which takes a very long time.

The device does accept at least 133 bytes w/ response and 300 bytes w/out response and it functions just fine with the manufacturer supplied mobile app. ( Cypress )

( On Android at least ) I did try upping the MTU using `requestMTU` but still does seem to chop off the data that I am sending. I have also tried using `write` and `writeWithoutResponse` with a varying combination of maxByteSize and in the end, the command that checks for each row fails.

Is this a BLE limit or is there a trick to sending +20 bytes of data?

**Comment by marcosinigaglia** [comment]
Hi, the only way in Android is changing the MTU, try to understand why the `requestMTU` is not working.
